### PR TITLE
help XXX will show the MESSAGE_USAGE string for the command

### LINKED ISSUE #120: `help XXX` will show the `MESSAGE_USAGE` string for the command

### PR DESCRIPTION
**Pull request by MAOXIONGKAI**
Closes #120, enhance `help` command with more flexibility, things to take note:
1. `help <Command Keyword>` displays `MESSAGE_USAGE` in the command class whose command word field is `<Command Keyword>`, hence it is the responsibility of command class implementer to change `MESSAGE_USAGE` so that `help` command displays accurate message
image
2. `help` displays both `FULL_HELP_MESSAGE` on the UI and pops up the help window with the same message. As shown in the screenshot below. However, personally I feel only one should be kept, what do you guys think? If yes, which one should I keep?
image
4. To efficiently format `FULL_HELP_MESSAGE` and reduce the burden of having to constantly manually update this after every command update, I have implemented a method called `getOrderedListString` which accepts an array and displays the element in order list string. The `help` command itself keeps an array of all command usage strings and `FULL_HELP_MESSAGE` gets the formatted string from this array. Hence, in the future if there's any new command, just update the array will keep the `FULL_HELP_MESSAGE` updated.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/HelpCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    /**
+     * Parses the given {@code String} of arguments in the context of the HelpCommand
+     * and returns a HelpCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    @Override
+    public HelpCommand parse(String keyword) throws ParseException {
+        String trimmedLowercaseKeyword = keyword.trim().toLowerCase();
+        if (!trimmedLowercaseKeyword.isEmpty()
+            && !trimmedLowercaseKeyword.equals(AddCommand.COMMAND_WORD)
```
Comment (weiseng18) [inline src/main/java/seedu/address/logic/parser/HelpCommandParser.java:29]: maybe can add `&& !trimmedLowercaseKeyword.equals(HelpCommand.COMMAND_WORD)` to allow `help help` to be a valid command

File: src/main/java/seedu/address/logic/commands/HelpCommand.java [hunk 2]
```diff
@@ -9,13 +15,76 @@ public class HelpCommand extends Command {
 
     public static final String COMMAND_WORD = "help";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Shows program usage instructions.\n"
+    public static final String MESSAGE_USAGE = COMMAND_WORD + " <Command Keyword>"
```
Comment (lithops-zty) [inline src/main/java/seedu/address/logic/commands/HelpCommand.java:18]:
```suggestion
    public static final String MESSAGE_USAGE = COMMAND_WORD + " COMMAND_KEYWORD"
```
for consistency

**Review by lithops-zty — COMMENTED** [review]
lgtm!

File: src/main/java/seedu/address/logic/parser/HelpCommandParser.java (hunk 1 above)
Comment (MAOXIONGKAI) [reply src/main/java/seedu/address/logic/parser/HelpCommandParser.java:29]: Very good spot! Thank you👍 @weiseng18 I just realized that I leave this thing out and caused `help help` command to fail, the test cases didn't catch this error either so I have fixed them also. The error message for `ParseException` is also wrong(I didn't use the `INVALID_HELP_KEYWORD` message I come up with lol), but now everything should be fine. Thank you!

**Review by lithops-zty — APPROVED** [review]
👍

**Pull request merged**